typescript/src/bitcoin.ts

```diff
@@ -573,39 +554,49 @@ export function hashLEToBigNumber(hash: Hex): BigNumber {
  *        unprefixed hex string (without 0x prefix).
  * @param witness - If true, a witness public key hash will be encoded and
  *        P2WPKH address will be returned. Returns P2PKH address otherwise
- * @param network - Network the address should be encoded for.
+ * @param bitcoinNetwork - Network the address should be encoded for.
  * @returns P2PKH or P2WPKH address encoded from the given public key hash
  * @throws Throws an error if network is not supported.
  */
 export function encodeToBitcoinAddress(
   publicKeyHash: string,
   witness: boolean,
-  network: BitcoinNetwork
+  bitcoinNetwork: BitcoinNetwork
 ): string {
-  const buffer = Buffer.from(publicKeyHash, "hex")
-  const bcoinNetwork = toBcoinNetwork(network)
+  const hash = Buffer.from(publicKeyHash, "hex")
+  const network = toBitcoinJsLibNetwork(bitcoinNetwork)
   return witness
-    ? bcoin.Address.fromWitnessPubkeyhash(buffer).toString(bcoinNetwork)
-    : bcoin.Address.fromPubkeyhash(buffer).toString(bcoinNetwork)
+    ? payments.p2wpkh({ hash, network }).address!
+    : payments.p2pkh({ hash, network }).address!
 }
 
 /**
  * Decodes P2PKH or P2WPKH address into a public key hash. Throws if the
  * provided address is not PKH-based.
- * @param address - P2PKH or P2WPKH address that will be decoded.
+ * @param bitcoinAddress - P2PKH or P2WPKH address that will be decoded.
+ * @param bitcoinNetwork - Bitcoin network.
  * @returns Public key hash decoded from the address. This will be an unprefixed
  *        hex string (without 0x prefix).
  */
-export function decodeBitcoinAddress(address: string): string {
-  const addressObject = new bcoin.Address(address)
+export function decodeBitcoinAddress(
+  bitcoinAddress: string,
+  bitcoinNetwork: BitcoinNetwork
```

Comment:
I had to add a new argument (Bitcoin network).
The underlying `bitcoinjs-lib` library requires it as it checks if the address matches the network.
In theory, we could avoid adding the Bitcoin network argument by deducing the network type from the provided address (in `mainnet` addresses start with `bc1`, `1`, `3` as opposed to `testnet`).
I would keep the Bitcoin network argument as passing the network type is how the library was designed.